libpod/image/image.go

```diff
@@ -247,15 +247,7 @@ func (i *Image) getLocalImage() (*storage.Image, error) {
 			return img.image, nil
 		}
 	}
-	hasReg, err := i.hasRegistry()
-	if err != nil {
-		return nil, errors.Wrapf(err, imageError)
-	}
 
-	// if the input name has a registry in it, the image isnt here
-	if hasReg {
-		return nil, errors.Errorf("%s", imageError)
-	}
```

Comment:
Moving this looks incorrect; if the user says `my-registry.example.com/foo/bar`, the code should not even _consider_ trying `localhost/my-registry.example.com/foo/bar`.

Isn’t this just `hasRegistry` being incorrect about `foo/bar` names? One implementation of that has been fixed in https://github.com/containers/libpod/pull/1387; it seems to me that `Image.hasRegistry` should be modified to share code with `decompose`, or should be replaced by `decompose`, or (least preferred) should be fixed similarly to `decompose`.

Reply:
In fact `decompose` has already been called above! Please nuke `Image.hasRegistry` entirely, then.

Reply:
yes, hasRegistry does not correctly figure this out.  I'll rework with your comments in mind.

Reply:
better?